src/Broadway/Domain/DomainMessage.php

```diff
@@ -131,4 +140,15 @@ public function andMetadata(Metadata $metadata)
 
         return new DomainMessage($this->id, $this->playhead, $newMetadata, $this->payload, $this->recordedOn);
     }
+
+    /**
+     * @param string $type
+     * @return $this
+     */
+    public function andType($type)
+    {
+        $this->type = $type;
```

Comment:
A new optional parameter could be added to DomainMessage class called "type" for example. That would make the creation of a new DomainMessage, like with the andMetadata decorator, possible